client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/gateway/fake/RestaurantFakeGateway.kt

```diff
@@ -62,7 +62,30 @@ class RestaurantFakeGateway : IRestaurantGateway {
         }
     }
 
-    override suspend fun createRestaurant(restaurant: NewRestaurantInfo): Restaurant {
+    override suspend fun getRestaurantById(id: String): Restaurant {
+        return restaurants.find { it.id == id }?.toEntity() ?: throw UnknownError()
```

Comment:
using first instead of find would through `NoSuchElementException` if it didn't find the first element
it's better than throwing unknown error if it was null